### PR TITLE
Added parameters --ca-directory and --ca-filepath for secured download

### PR DESCRIPTION
**Pull request by OleksandrChaika**

### CODE REVIEW

File: src/Context.h [hunk 1]
```diff
@@ -55,11 +55,12 @@ namespace sua {
...
         std::shared_ptr<IMqttProcessor>         mqttProcessor;
         std::shared_ptr<IBundleChecker>         bundleChecker;
-        std::string                             updatesDirectory    = "/data/selfupdates";
-        std::string                             tempFileName        = "/temp_file";
-        std::string                             certificateFileName = "/etc/ssl/certs/ca-certificates.crt";
-        bool                                    downloadMode        = true;
-        bool                                    fallbackMode        = false;
+        std::string                             updatesDirectory = "/data/selfupdates";
+        std::string                             tempFileName     = "/temp_file";
+        std::string                             caPath           = "/etc/ssl/certs";
```
Comment (OlavHerbst) [inline src/Context.h]: suggestion again: use a global define or const string to specify it only once in src (Context, help text, ...)

File: src/main.cpp [hunk 2]
```diff
@@ -51,8 +51,11 @@ SUA_SERVER    sets and overrides MQTT server address to connect
                 (default is '/data/selfupdates')
 -s, --server    MQTT broker server to connect, has precedence over SUA_SERVER environment variable
                 (default is 'tcp://mosquitto:1883')
--c, --ca        path to certificate to verify connection with bundle server
-                (default is '/etc/ssl/certs/ca-certificates.crt')
+-c, --ca-path   specifies directory containing CA certificates to verify connection with bundle server
+                (default is '/etc/ssl/certs')
```
Comment (OlavHerbst) [inline src/main.cpp]: "specifies" is obsolete, removing it would decrease the line length

File: src/main.cpp [hunk 3]
```diff
@@ -51,8 +51,11 @@ SUA_SERVER    sets and overrides MQTT server address to connect
...
+-f, --ca-file   path to certificate to verify connection with bundle server
+                if set supersedes value set by --ca-path
+                (default is '')
```
Comment (OlavHerbst) [inline src/main.cpp]: "to certificate" could be read misleading as infinitve. 

suggestion:
"file path to the CA certificate to verify connection with bundle server"
"If set it supersedes the CA directory in --ca-path. (default is unset)"

File: src/Context.h [hunk 4]
```diff
@@ -55,11 +55,12 @@ namespace sua {
...
+        std::string                             caFile           = "/etc/ssl/certs/ca-certificates.crt";
```
Comment (OlavHerbst) [inline src/Context.h]: Init. of updatesDirectory, caPath and caFile could be empty, as these are is set in main().

File: src/main.cpp (hunk 3 above)
Comment (OlavHerbst) [inline src/main.cpp]: The usage of "path", "directory" and "file" in help text and member names could be misleading.
ca-path: "path" => a "directory"
ca-file: "file" => a "path"    ( ? here is "path" != ca-"path"). Here better "File path" ?

**Review by OlavHerbst — APPROVED** [review]
only some phrasing minors

File: src/main.cpp (hunk 3 above)
Comment (OleksandrChaika) [reply src/main.cpp]: Fixed

File: src/Context.h (hunk 4 above)
Comment (OleksandrChaika) [reply src/Context.h]: Fixed

File: src/main.cpp (hunk 3 above)
Comment (OleksandrChaika) [reply src/main.cpp]: Fixed

File: src/main.cpp (hunk 2 above)
Comment (OleksandrChaika) [reply src/main.cpp]: Fixed

File: src/Context.h (hunk 1 above)
Comment (OleksandrChaika) [reply src/Context.h]: Fixed

File: src/main.cpp [hunk 5]
```diff
@@ -30,42 +31,43 @@
...
+                   or 'dummy' for neither download nor installation
+                   (default is '{}')
+-p, --path         path where downloaded update bundles will be stored
+                   (default is '{}')
+-s, --server       MQTT broker server to connect, has precedence over SUA_SERVER environment variable
+                   (default is '{}')
+-c, --ca-directory a directory containing CA certificates to verify connection with bundle server
+                   (default is '{}')
+-f, --ca-filepath  file path to the CA certificate to verify connection with bundle server
+                   if set it supersedes the CA directory in --ca-directory (default is unset)
```
Comment (OlavHerbst) [inline src/main.cpp]: "(default..." is doubled

**Review by OlavHerbst — APPROVED** [review]
good idea to separate all default values
ok

File: src/main.cpp (hunk 5 above)
Comment (OleksandrChaika) [reply src/main.cpp]: Fixed

**Review by OlavHerbst — APPROVED** [review]
ok

**Pull request merged**